lib/http.dart

```diff
@@ -32,8 +32,10 @@ export 'src/streamed_response.dart';
 /// the same server, you should use a single [Client] for all of those requests.
 ///
 /// For more fine-grained control over the request, use [Request] instead.
-Future<Response> head(Uri url, {Map<String, String>? headers}) =>
-    _withClient((client) => client.head(url, headers: headers));
+Future<Response> head(Uri url,
+        {Map<String, String>? headers, Duration? timeout}) =>
```

Comment:
Document the parameter.

What happens when the timeout duration has passed? I'm *expecting* the returned future to complete with a `TimeoutException`. If so, that's a reasonable default, and there isn't really a good alternative.

Reply:
Yes I think completing with a `TimeoutException` is the most useful API.

If it turns out that it is more common to want to ignore the timeout and have a `Future` that doesn't complete instead we could drop a utility for it in `package:async`

```dart
extension IgnoreTimeout<T> on Future<T> {
  Future<T> ignoreTimeout() => catchError((_) {}, test:(e) => e is TimeoutException);
}
```

Reply:
I don't think anyone ever wants a future which doesn't complete.
(They might think they do, but then they end up doing `try { await something; } finally { never runs!!!?! }` and change their mind).

Reply:
Added docs.